### PR TITLE
actually start glauth server

### PR DESCRIPTION
**Pull request by butonic**
To test install the oc10 [graphapi app](https://github.com/owncloud/graphapi) in an oc instance, then replace the url with your instance in this line
```
bin/ocis-glauth --log-level debug server --backend-server https://cloud.example.com
```
query the givenname of all your users like this
```
ldapsearch -x -H ldap://localhost:9125 -b dc=owncloud,dc=com -D "cn=<loginname>,dc=owncloud,dc=com" -W givenName
```
replace `<loginname>` with you login, eg admin ... ldapsearch will ask for your password and you should see the users displayname / givenname. try fetching groups with
```
 ldapsearch -x -H ldap://localhost:9125 -b dc=owncloud,dc=com -D "cn=<loginname>,dc=owncloud,dc=com" objectclass=posixaccount -W
```

:rocket: 

This PR
- [x] adds config flags for glauth ldap ports and the backend
- [ ] add config options for the http port of glauth
- [x] add config options for the ldaps cert and key
- [x] by default uses https://demo.owncloud.com as the data store
  - [ ] but that does not have the graphapi app installed ... 
- [x] when killing the server it neds to be killed hard and exits with
```
2020-02-25T14:55:41+01:00 INF Shutting down server service=glauth transport=http
2020-02-25T14:55:41+01:00 INF Shutting down server service=glauth transport=debug
^Cpanic: send on closed channel

goroutine 19 [running]:
os/signal.process(0x12a6a20, 0xc000264308)
        /usr/lib/go/src/os/signal/signal.go:227 +0x189
os/signal.loop()
        /usr/lib/go/src/os/signal/signal_unix.go:23 +0x3e
created by os/signal.init.0
        /usr/lib/go/src/os/signal/signal_unix.go:29 +0x41
```
  may need some more tweaking upstream to cleanly kill the server by using a context ...
- [ ] when not using the graphapi I only see the same user over and over in the list ...
```
ldapsearch -x -H ldap://localhost:9125 -b ou=users,dc=owncloud,dc=com -D "cn=jfd,ou=users,dc=owncloud,dc=com" -W 
Enter LDAP Password: 
# extended LDIF
#
# LDAPv3
# base <ou=users,dc=owncloud,dc=com> with scope subtree
# filter: (objectclass=*)
# requesting: ALL
#

# test2, users, owncloud.com
dn: cn=test2,ou=users,dc=owncloud,dc=com
cn: test2
uid: test2
objectClass: posixAccount
description: test2 from ownCloud

# test2, users, owncloud.com
dn: cn=test2,ou=users,dc=owncloud,dc=com
cn: test2
uid: test2
objectClass: posixAccount
description: test2 from ownCloud

.
.
.

# test2, users, owncloud.com
dn: cn=test2,ou=users,dc=owncloud,dc=com
cn: test2
uid: test2
objectClass: posixAccount
description: test2 from ownCloud

# search result
search: 2
result: 0 Success
```
- [ ] when using the graphapi I don't seem to be able to list users
```
 ldapsearch -x -H ldap://localhost:9125 -b ou=users,dc=owncloud,dc=com -D "cn=jfd,ou=users,dc=owncloud,dc=com" -W  
Enter LDAP Password: 
# extended LDIF
#
# LDAPv3
# base <ou=users,dc=owncloud,dc=com> with scope subtree
# filter: (objectclass=*)
# requesting: ALL
#

# search result
search: 2
result: 1 Operations error

# numResponses: 1
```

### CODE REVIEW

File: pkg/command/server.go [hunk 1]
```diff
@@ -144,7 +170,7 @@ func Server(cfg *config.Config) *cli.Command {
 				}
 
 				gr.Add(func() error {
-					//return server.Run()
+					server.ListenAndServe()
```
Comment (DeepDiver1975) [inline pkg/command/server.go]:
```go
if err := server.listenAndServer; err != nill {
logger.Error(err)
}

Comment (DeepDiver1975) [reply pkg/command/server.go]: and return the err - afaik ....


if we do not log this err here we never know why the server is not starting (port already in use e.g.)

Comment (butonic) [reply pkg/command/server.go]: the ListenAndServe in glauth currently does notr return an error ...

Comment (DeepDiver1975) [reply pkg/command/server.go]: oh - we shall fix this then ....

Comment (butonic) [reply pkg/command/server.go]: fixed with https://github.com/glauth/glauth/pull/126

File: changelog/unreleased/initial-release.md [hunk 2]
```diff
@@ -1,6 +1,5 @@
-Change: Initial release of basic version
+Enhancement: Initial release of basic version
 
-Just prepare an initial basic version to serve a glauth world API that can be
-used by Phoenix or other extensions.
+Just prepare an initial basic version to provide a glauth service.
 
-https://github.com/owncloud/ocis-glauth/issues/7
+https://github.com/owncloud/ocis-glauth/issues/2
```
Comment (micbar) [inline changelog/unreleased/initial-release.md]: Can you either link to the PR or copy the long description to the issue?

**Review by micbar — APPROVED** [review]

**Pull request merged**